src/sys/unix.rs

```diff
@@ -2711,7 +2711,7 @@ impl crate::Socket {
         .map(|_| {
             let buf = &payload[..len as usize];
             // TODO: use `MaybeUninit::slice_assume_init_ref` once stable.
-            unsafe { &*(buf as *const [_] as *const [u8]) }.into()
+            unsafe { &*(buf as *const [u8]) }.into()
```

Comment:
The payload isn't uninitialized, so I don't think this is necessary.
```suggestion
            buf.to_vec()
```